### PR TITLE
readd support for changing SYSCONFDIR

### PR DESCRIPTION
**Pull request by MartinNowak**
- used by homebrew dmd recipe

### CODE REVIEW

File: src/posix.mak [hunk 1]
```diff
@@ -280,6 +281,8 @@ TK_SRC = \
 	$(TK)/filespec.h $(TK)/mem.h $(TK)/list.h $(TK)/vec.h \
 	$(TK)/filespec.c $(TK)/mem.c $(TK)/vec.c $(TK)/list.c
 
+STRING_IMPORT_FILES = verstr.h SYSCONFDIR
```
Comment (yebblies) [inline src/posix.mak]: Please give it a different name to the make variable.

**Comment by John-Colvin** [comment]
this would be great, it's a pain having the homebrew dmd head version not working.

File: src/posix.mak (hunk 1 above)
Comment (MartinNowak) [inline src/posix.mak]: What do you mean?

Comment (yebblies) [inline src/posix.mak]: I mean you've got a file named `SYSCONFDIR` and a make variable called `SYSCONFDIR` and it's a little confusing.

Comment (WalterBright) [inline src/posix.mak]: I agree with @yebblies 
makefiles are hard enough to understand without the dual use of the name.

Comment (MartinNowak) [inline src/posix.mak]: Whatever, renamed to SYSCONFDIR.imp.

**Comment by yebblies** [comment]
No idea what's going on here, but I'm sure you'll fix it if it breaks something.

**Comment by yebblies** [comment]
Auto-merge toggled on

File: src/posix.mak [hunk 2]
```diff
@@ -365,6 +369,8 @@ VERSION := $(addsuffix -devel$(if $(VERSION_GIT),-$(VERSION_GIT)),$(VERSION))
 endif
 $(shell test \"$(VERSION)\" != "`cat verstr.h 2> /dev/null`" \
 		&& printf \"$(VERSION)\" > verstr.h )
+$(shell test $(SYSCONFDIR) != "`cat SYSCONFDIR.imp 2> /dev/null`" \
```
Comment (CyberShadow) [inline src/posix.mak]: .gitignore plz

**Pull request merged**